Add a sphinx docs structure for hosting gmt examples (#1)
GenericMappingTools/gmt-examples: docs/contributing.rst

Two options for installing the dependencies are :ref:`pip <Pip Setup>` and :ref:`conda <Conda Setup>`.
Since Sphinx is a Python documentation generator, you will need a working Python
environment. For those who do not already have a working Python environment,
one option is to use the minimal installer for Conda `Miniforge <https://github.com/conda-forge/miniforge>`_
along with the :ref:`conda <Conda Setup>` instructions.

Pip Setup
^^^^^^^^^

These instructions rely on the `pip <https://pip.pypa.io/en/stable/>`_ package
installer, which can be used to install all dependencies except GMT.
Follow the `GMT Install Guide <https://github.com/GenericMappingTools/gmt/blob/master/INSTALL.md>`_
to install GMT. The Python dependencies can be installed using the
``requirements.txt`` from the base of the repository (you may wish to setup a
`virtual environment <https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/#creating-a-virtual-environment>`_ first):

Unix/macOS::

  python -m pip install -r requirements.txt

> 

Reviewer: ```suggestion
```

Windows::

  py -m pip install -r requirements.txt

Conda Setup
^^^^^^^^^^^

These instructions rely on the `conda <https://docs.conda.io/en/latest/>`_ package
manager. Run the following from the base of the repository to create a new conda
environment from the ``environment.yml`` file::

  conda env create

> 

Reviewer: ```suggestion
```

Before building the documentation, you have to activate the environment
(you'll need to do this every time you start a new terminal)::

  conda activate gmt-examples

Building the documentation
~~~~~~~~~~~~~~~~~~~~~~~~~~

Once you have :ref:`setup your environment <Setting up your environment>`, you can
build the documentation using::

  cd docs
  make html

Contributing New Examples
~~~~~~~~~~~~~~~~~~~~~~~~~

The source files for the gallery examples and tutorials are ``.rst`` files in
``docs/`` that generate one or more figures using the
`sphinx_gmt <https://www.generic-mapping-tools.org/sphinx_gmt/latest/)>`_
extension. To add a new gallery example or tutorial:

* If necessary, create a new sub-directory under ``docs/gallery/`` for a
  gallery example section (e.g., ``basemaps/`` or ``plot_embellishments/``).
* Create a new ``.rst`` file inside the appropriate sub-directory in
  ``docs/gallery/`` or ``docs/tutorials/``.
* Add a descriptive title and as much explanation as necessary.

> * Add hyperlinks to GMT modules using the ``gmt-module`` restructured text directive::

Reviewer: ```suggestion
* Add hyperlinks to GMT modules using the ``gmt-module`` reStructuredText directive::
```

  :gmt-module:`grdview`

* Add hyperlinks to GMT module options by appending lower-case ``#<option>`` to
  the ``gmt-module`` argument::

  :gmt-module:`grdview#q`

* Add as many figures as needed using the ``gmtplot`` directive:

  ::

>     .. gmtplot::
> 
>       gmt begin basemap png
>         gmt basemap -B -Rg -JH5c
>       gmt end show

Reviewer: reStructureText needs 3 or 4 leading whitespaces.
```suggestion
    .. gmtplot::

       gmt begin basemap png
         gmt basemap -B -Rg -JH5c
       gmt end show
```

  The figures will be placed after the source code in the built documentation
  by the ``sphinx_gmt`` extension.

* Add the file to the appropriate section in ``docs/index.rst`` using the following
  template:

  ::

>     ```bash
>       -  .. image:: _images/<file-name>-gmtplot-0.png
>           :target: gallery/<section>/<file-name>.html
>           :width: 80%
>           :align: center
> 
>        :doc:`gallery/<section>/<file-name>`
>     ```

Reviewer: ```suggestion
    ```bash
      -  .. image:: _images/<file-name>-gmtplot-0.png
            :target: gallery/<section>/<file-name>.html
            :width: 80%
            :align: center

         :doc:`gallery/<section>/<file-name>`
    ```
```
